sphinx/cmdline.py

```diff
@@ -191,23 +194,62 @@ def main(argv):
 
     # get paths (first and second positional argument)
     try:
-        srcdir = abspath(args[0])
-        confdir = abspath(opts.confdir or srcdir)
-        if opts.noconfig:
-            confdir = None
+        # Check src directory (first positional argument)
+        srcdir = path.abspath(args[0])
         if not path.isdir(srcdir):
             print('Error: Cannot find source directory `%s\'.' % srcdir,
                   file=sys.stderr)
             return 1
-        if not opts.noconfig and not path.isfile(path.join(confdir, 'conf.py')):
-            print('Error: Config directory doesn\'t contain a conf.py file.',
-                  file=sys.stderr)
-            return 1
+
+        # Check output directory (second positional argument)
         outdir = abspath(args[1])
         if srcdir == outdir:
             print('Error: source directory and destination directory are same.',
                   file=sys.stderr)
             return 1
+
+        # Check config parameters (-c option and -C flag)
+        confdir = None
+        conffilename = None
+        if opts.noconfig:
+            if not opts.confpath:
+                confdir = srcdir
```

Comment:
If `opts.noconfig` is specified, `confdir` should have None value.
Are you tested the option at local?

(unfortunately, there are no testcases for `sphinx.cmdline`. So we have to test it carefully)